packaging/bundle-validation/validate.sh

```diff
@@ -0,0 +1,114 @@
+    #put config files in correct place
+    cp $HIVE_DATA/spark-defaults.conf $SPARK_HOME/conf/
+    cp $HIVE_DATA/hive-site.xml $HIVE_HOME/conf/
+    ln -sf $HIVE_HOME/conf/hive-site.xml $SPARK_HOME/conf/hive-site.xml
+    cp $DERBY_HOME/lib/derbyclient.jar $SPARK_HOME/jars/
+
+    $DERBY_HOME/bin/startNetworkServer -h 0.0.0.0 &
+    $HIVE_HOME/bin/hiveserver2 &
+    echo "::warning::validate.sh hive setup complete. Testing"
+    $SPARK_HOME/bin/spark-shell --jars $JAR_DATA/spark.jar < $HIVE_DATA/validate.scala
+    if [ "$?" -ne 0 ]; then
+        echo "::error::validate.sh failed hive testing)"
+        exit 1
+    fi
+    echo "::warning::validate.sh hive testing succesfull. Cleaning up hive sync"
+    #remove config files
+    rm -f $SPARK_HOME/jars/derbyclient.jar
+    unlink $SPARK_HOME/conf/hive-site.xml
+    rm -f $HIVE_HOME/conf/hive-site.xml
+    rm -f $SPARK_HOME/conf/spark-defaults.conf
+}
+
+test_utilities_bundle () {
+    echo "::warning::validate.sh running deltastreamer"
+    $SPARK_HOME/bin/spark-submit --driver-memory 8g --executor-memory 8g \
+    --class org.apache.hudi.utilities.deltastreamer.HoodieDeltaStreamer \
+    $JARS_BEGIN $JARS_ARG \
+    --props $UTILITIES_DATA/parquet-dfs-compact.props \
+    --schemaprovider-class org.apache.hudi.utilities.schema.FilebasedSchemaProvider \
+    --source-class org.apache.hudi.utilities.sources.ParquetDFSSource \
+    --source-ordering-field date_col --table-type MERGE_ON_READ \
+    --target-base-path file://${OUTPUT_DIR} \
+    --target-table ny_hudi_tbl  --op UPSERT
+    echo "::warning::validate.sh done with deltastreamer"
+
+    OUTPUT_SIZE=$(du -s ${OUTPUT_DIR} | awk '{print $1}')
+    if [[ -z $OUTPUT_SIZE || "$OUTPUT_SIZE" -lt "1000" ]]; then
+        echo "::error::validate.sh deltastreamer output folder is much smaller than expected)" 
+        exit 1
+    fi
+
+    echo "::warning::validate.sh validating deltastreamer in spark shell"
+    SHELL_COMMAND="$SPARK_HOME/bin/spark-shell --jars $JARS_ARG $SHELL_ARGS -i $COMMANDS_FILE"
+    echo "this is the shell command: $SHELL_COMMAND"
+    LOGFILE="$WORKDIR/submit.log"
+    $SHELL_COMMAND >> $LOGFILE 
+    if [ "$?" -ne 0 ]; then
+        SHELL_RESULT=$(cat $LOGFILE | grep "Counts don't match")
+        echo "::error::validate.sh $SHELL_RESULT)"
+        exit 1
+    fi
+    echo "::warning::validate.sh done validating deltastreamer in spark shell"
+}
+
+
+# run_hive_sync
+# if [ "$?" -ne 0 ]; then
+#     exit 1
+# fi
+
+SHELL_ARGS=$(cat $UTILITIES_DATA/shell_args)
+
+echo "::warning::validate.sh testing utilities bundle"
+JARS_BEGIN=""
+JARS_ARG=$JAR_DATA/utilities.jar
+OUTPUT_DIR=/tmp/hudi-deltastreamer-ny/
+COMMANDS_FILE=$UTILITIES_DATA/commands.scala
+test_utilities_bundle
+if [ "$?" -ne 0 ]; then
+    exit 1
+fi
+echo "::warning::validate.sh done testing utilities bundle"
+
+echo "::warning::validate.sh testing utilities slim bundle"
+JARS_BEGIN="--jars"
+JARS_ARG="$JAR_DATA/spark.jar $JAR_DATA/utilities-slim.jar"
```

Comment:
JARS_BEGIN and JAR_ARG are a bit weird. you can set `MAIN_JAR` to be always using utilities bundle and `JARS_OPT='--jars xxx'` or empty string. MAIN_JAR can only be set to 1 jar and JARS_OPT can be multiple comma-delimited jars